src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp

```diff
@@ -44,6 +44,37 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
         return;
     }
 
+    auto prefix = MetaServiceUtils::indexPrefix(space);
+    std::unique_ptr<kvstore::KVIterator> checkIter;
+    auto checkRet = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &checkIter);
+    if (checkRet != kvstore::ResultCode::SUCCEEDED) {
+        resp_.set_code(to(checkRet));
+        onFinished();
+        return;
+    }
+
+    while (checkIter->valid()) {
+        auto val = checkIter->val();
+        auto item = MetaServiceUtils::parseIndex(val);
+        if (item.get_schema_id().getType() != nebula::cpp2::SchemaID::Type::edge_type) {
+            continue;
+        }
+
+        for (size_t i = 0; i < fieldNames.size(); i++) {
+            if (fieldNames[i] != item.get_fields()[i].get_name()) {
+                break;
+            }
+
+            if (i == fieldNames.size() - 1) {
+                LOG(ERROR) << "Edge Index " << indexName << " have existed";
+                resp_.set_code(cpp2::ErrorCode::E_EXISTED);
+                onFinished();
+                return;
+            }
+        }
+        checkIter->next();
+    }
+
```

Comment:
A question， Do we need to keep naming unique between tag index and edge index like tag and edge naming?

Reply:
Actually, I don't know why the index name is duplicated.